sympy/physics/control/lti.py

```diff
@@ -1702,9 +2043,13 @@ def __new__(cls, arg):
         for row_index, row in enumerate(arg):
             temp = []
             for col_index, element in enumerate(row):
-                if not isinstance(element, (TransferFunction, Series, Parallel)):
+                try:
+                    if not element.is_SISO:
+                        raise TypeError("MIMO Series/Parallel object found as the element of "
+                        f"TransferFunctionMatrix at index {(row_index, col_index)}.")
+                except AttributeError:
                     raise TypeError("Incompatible type found as the element of "
-                    f"TransferFunctionMatrix at index {(row_index, col_index)}.")
+                        f"TransferFunctionMatrix at index {(row_index, col_index)}.")
```

Comment:
I see. This is what you meant. 
More like, 
```python
                try:
                    if not element.is_SISO:
                        raise TypeError("MIMO Series/Parallel object found as the element of "
                        f"TransferFunctionMatrix at index {(row_index, col_index)}.")
                except AttributeError as err:
                    raise TypeError("Incompatible type found as the element of "
                        f"TransferFunctionMatrix at index {(row_index, col_index)}.") from err
```

Reply:
We shouldn't catch AttributeError. AttributeError is usually raised because of bugs in the code. Instead just use `getattr`:
```python
if getattr(element, 'is_SISO', False):
    ...
```
Why are you checking if this attribute exists anyway?

It would be better to check the type of the arguments and then ensure that all acceptable types have the `is_SISO` attribute:
```python
rows_list = arg
var = rows_list[0][0]  # Can there be zero rows or columns?

values = set.union(*rows_list)
if not all(isinstance(v, SomeCls) for v in values):
    raise TypeError("Inputs should be SomeCls")
if not all(v.is_SISO for v in values):
    raise TypeError("Inputs should be SISO")

expr_mat = ImmutableMatrix([[v.to_expr() for v in row] for row in rows_list])
```
As I mentioned somewhere else there should be a class `SomeCls` that you can test for here.

Reply:
@namannimmo10, should I reintroduce the `TransferFunctionExpr` class, or you prefer some other name?

Reply:
I think that TransferFunctionExpr sounds like a restrictive superclass that only applies to transfer function like instances.
We could have other objects like StateSpace (to appear), so a common name like ControlSystems could be appropriate for separating the objects from this library from the rest of the sympy.

But I don’t know if others prefer duck typing (like SISO statespace can also be recognized as something like SISO transferfunction) or they should be in a completely different and incompatible objects.

But I believe that this is fine since I think that we can have many references how these are taxonomically organized.